src/main/java/org/broadinstitute/hellbender/engine/SlidingWindowWalker.java

```diff
@@ -0,0 +1,185 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.engine.filters.CountingReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilterLibrary;
+import org.broadinstitute.hellbender.engine.filters.WellformedReadFilter;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * An SlidingWindowWalker is a tool that processes a single window over the genome/intervals at a time, with the ability
+ * to query optional overlapping sources of reads, reference data, and/or variants/features.
+ *
+ * The windows are constructed over for each interval with a concrete window and step size, and padding overlapping
+ * sources if requested
+ *
+ * SlidingWindow authors must implement the {@link #apply}, {@link #getWindowSize}, {@link #getWindowStep} and {@link
+ * #getWindowPadding} methods, and may optionally implement {@link #onTraversalStart} and/or {@link #onTraversalSuccess}.
+ *
+ * @author Daniel Gómez-Sánchez (magicDGS)
+ */
+public abstract class SlidingWindowWalker extends GATKTool {
+
+    @Argument(fullName = "disable_all_read_filters", shortName = "f", doc = "Disable all read filters", common = false, optional = true)
+    public boolean disableAllReadFilters = false;
+
+    protected final static String WINDOW_SIZE_ARGUMENT = "windowSize";
+
+    protected final static String WINDOW_STEP_ARGUMENT = "windowStep";
+
+    protected final static String WINDOW_PADDING_ARGUMENT = "windowPadding";
```

Comment:
About the arguments, I'm not sure if in this case fit. One of the problems is that some tools implementing a `SlidingWindowWalker` won't like to expose to the final user some parameters. One of the use case that I can imagine is a window walker that does not require any padding; or a walker that requires exactly the same window size and step (as in the case of the `AssemblyRegionWalker`).

Is there a way of hide the `|@Argument` annotations in classes that extends this tool?